### PR TITLE
Terms checkbox

### LINKED ISSUE #385: Add checkbox for terms agreement
It might be more legally enforceable if the user clicks a box saying they agree to the terms rather than agreeing by default by clicking register

### PR DESCRIPTION
**Pull request by jsullivan5**
This PR closes #385 #389 

#### What does this PR do?

* This adds a checkbox to accept terms of use.
* Register button is disabled until it is checked.
* agreeTerms is added to the user model and added to the database as boolean when user is registered.  (Not sure if this is necessary)
* Refactors change handler functions to use generic `handleInputChange` with name properties on the inputs
* Changes dynamic classNames in Register component to template literals to fix linter warning.
* Updates lander snapshot test to account for new html markup

### CODE REVIEW

**Comment by eric-yates** [comment]
I'd personally put the text label on the right side of the checkbox. That is consistent with typical formatting guidelines. Also, could you add an X to close the terms pop-up window?

I'm not knowledgeable enough to know whether the changes you made are well done. I'll assign Nathan to review

File: client/src/components/Register.js [hunk 1]
```diff
@@ -39,21 +38,21 @@ class Register extends React.Component {
...
-    this.setState({ isMusician: ev.target.checked });
+    this.setState({
+      [name]: value
+    });
   }
 
   handleFormSubmit(ev) {
     ev.preventDefault();
-    this.props.registerRequest(this.state.email, this.state.password, this.state.isMusician);
+    const { email, password, isMusician, agreeTerms } = this.state;
```
Comment (nlkluth) [inline client/src/components/Register.js:53]: Would it make sense to have these inputs be uncontrolled instead of using state?

**Review by nlkluth — APPROVED** [review]

File: client/src/components/Register.js (hunk 1 above)
Comment (jsullivan5) [reply client/src/components/Register.js:53]: Hi Nathan!  The render of the Register component depends on agreeTerms and isMusician for disabling the submit button and showing different content.  I find it easier to manage those relationships when the inputs are controlled.  Let me know if you think that's overkill. 😀

Comment (nlkluth) [reply client/src/components/Register.js:53]: Cool no worries. LGTM

**Comment by jsullivan5** [comment]
@eric-yates Good call on the checkbox and label.  That has been fixed.  Did you see the close button at the bottom of the terms?  I can get an X style close button in the top right if we want.  I feel like users are used to scrolling to the bottom of these to close them though.

**Comment by eric-yates** [comment]
@jsullivan5 Looks good on the checkbox. I did not see the close button at the bottom, so disregard the X close button. It'd be best to have people scrolling to the bottom to close anyways. Thanks and great work!

**Pull request merged**